### DISCUSSION TITLE
when to halcompile and when to copy configs

### DISCUSSION THREAD

**Discussion by jschoch** [multigcs/LinuxCNC-RIO#40] [Ideas]
It may not be clear for a newbie when you need to halcompile or not.  

I assume you want to keep the rio module clean, however it simply doesn't connect if you change your config (add enc_z for example) without any warnings.  I wonder if it is better to detect and throw some warnings if your config can't be managed with the currently compiled module?  Alternatively it may be useful to document which plugins need a halcompile and or .hal/.ini change.

Happy to help just want to discuss options first.

**Comment by multigcs** [comment]
halcompile is allways needed if the interface has changed (between FPGA and Host)

i allways run halcompile after changes because it costs max 2 seconds  :)

**Comment by jschoch** [comment]
are you not open to stubbing out the interfaces so that a runtime error can be generated?

**Comment by multigcs** [comment]
for the new version i have planned to replace the msgid (header) with a checksum.
the check-sum is then generated by the interface description.
